src/main/java/com/ixortalk/autoconfigure/oauth2/auth0/mgmt/api/Auth0ManagementAPI.java

```diff
@@ -184,7 +204,10 @@ public void removeRolesFromUser(String userId, List<String> roleIds) {
         }
     }
 
-    @CacheEvict(cacheNames = {AUTH_0_USER_CACHE, AUTH_0_ROLE_USER_CACHE}, allEntries = true)
+    @Caching(evict = {
+            @CacheEvict(cacheNames = AUTH_0_USER_CACHE, key = "#email"),
```

Comment:
Related to previous comment, user gets evicted here by email, but the entry "by id" remains in the cache.

Reply:
I separated the 2 caches. As the user will be created here, the cache probably won't contain the user yet. I'll change it to deleting only the "user by email" cache, as the id is not yet known at this point.